Applications/FileIO/PetrelInterface.h

```diff
@@ -39,6 +39,11 @@ class PetrelInterface final
                     std::string &unique_model_name,
                     GeoLib::GEOObjects* obj);
 
+    PetrelInterface(PetrelInterface const& other) = delete;
+    PetrelInterface(PetrelInterface&& other) = delete;
+    PetrelInterface& operator=(PetrelInterface const&) = delete;
+    PetrelInterface& operator=(PetrelInterface&&) = delete;
```

Comment:
Just curious, why explicitly delete the ctor's/assignments?

Reply:
The class 'PetrelInterface' contains pointer to allocated memory. So a correct implementation would be necessary rather than a compiler generated. I thought it is better to mark ctor's/assignments as delete.